testinfra/test_ami_nix.py

```diff
@@ -321,14 +329,103 @@ def is_healthy(host, instance_ip, ssh_identity_file) -> bool:
 
         for service, check in health_checks:
             try:
-                cmd = check(host)
-                if cmd.failed is True:
-                    logger.warning(f"{service} not ready")
-                    return False
-            except Exception:
+                if service == "postgres":
+                    # For PostgreSQL, we need to check multiple things
+                    systemd_status, socket_check, pg_isready = check(host)
+                    
+                    if systemd_status.failed:
+                        logger.error("PostgreSQL systemd service is not active")
+                        logger.error(f"systemd status: {systemd_status.stdout}")
+                        logger.error(f"systemd error: {systemd_status.stderr}")
+                        
+                        # Get detailed systemd status
+                        logger.error("Detailed systemd status:")
+                        host.run("sudo systemctl status postgresql -l --no-pager")
+                        
+                        # Check init script logs
+                        logger.error("Init script logs:")
+                        host.run("sudo journalctl -u cloud-init --no-pager")
```

Comment:
Q: Can this logic be tested locally? Or does it rely on a component outside of this repo?

Reply:
@steve-chavez it's a high priority for me to be able to test this locally, but as of this moment it cannot. I don't like this state of affairs and so will be finding some way to change it ASAP. For today, I am stuck doing it this way.

Reply:
(it relies on creating an AWS ami, and running an ec2 instance. In the RFC I am working on, it will recognize if it is trying to be run in aws or in a local ubuntu vm, and switch context and run)

Reply:
It will also have a way to create a base AMI nightly, boot a machine based off of the latest nightly available, and make the changes that need to be made only. This will drastically speed up all AMI building operations

Reply:
@steve-chavez over here I came very close to being able to run this locally, although it really in that case is just still running on aws which does not solve the problem fully. I'd rather be able to run fully local, or in ci. But will need to do some work to get this our repo to achieve that. But it's going to happen relatively soon.

Reply:
@samrose All good. Thanks for elaborating.

Reply:
@steve-chavez I am actually creating a PR to this PR, to try and have an immediate way to run this locally.

Reply:
It's too much work to piggy back on this. But we're going to get into it soon